agent_api_rest/src/identity/profiles/mod.rs

```
@@ -0,0 +1,78 @@
+use crate::error::IntoApiErrorExt;
+use crate::handlers::{command_handler, query_handler};
+use crate::utils::serde_explicit_null;
+use agent_identity::profile::aggregate::Source;
+use agent_identity::profile::command::ProfileCommand;
+use agent_identity::state::IdentityState;
+use agent_identity::state::{query_profile, PROFILE_ID};
+use agent_shared::config::Logo;
+use axum::{
+    extract::State,
+    response::{IntoResponse, Response},
+    Json,
+};
+use http_api_problem::ApiError;
+use hyper::StatusCode;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
```

Should we `#[serde(deny_unknown_fields)]` to throw an error when unknown keys are supplied during PATCH? Do you think this adds more explicitness or is it not necessary?
Reason I am asking this, is because I am currently getting a clean `200 OK` with no further response body when supplying `display_name` instead of `displayName`, so I thought the update was successful.

---

The real problem isn't about whether we ignore unknown fields (which is standard anyway), it's the inconsistency in the API.

Going with 100% snake_case would be the path of least resistance since it's the Rust default. But you're right, camelCase is probably friendlier for our API users in the long run. The main drawback is that it's more error-prone for us; we'd have to be diligent about testing and remember to handle external types, which won't get converted automatically. It just adds a bit more overhead.

So imo best would be: keep ignoring unknown fields + make sure to use camelCase consistently (+ have automatically generated API reference? + have automatically generated wasm/typescript bindings?)